### ISSUE TITLE
Commit message format in the wiki

### ISSUE THREAD

**Issue by mohrezaei** [eclipse/eclipse-collections#972]
The wiki has new language about the commit message, which is neither consistent with the history of this project, nor represents a consensus from the committers that I know of. The language says:
```Commit message must begin with `Closes #<issue-number>` ```

This is bad on multiple fronts.

1) It obscures the actual message as it appears in front.
2) It references something outside the repository.

Code survives. Tooling around code does not. This very same code has been through many different version control systems, and bug tracking systems. If we had embedded references to old bug #'s in the actual repository, it would've been less than useless now.

To be clear, I think it's a good idea to mention the Github Issue in the Github Pull Request. They go together. You can click on them when you're looking at them. Their lifecycle is likely the same.

However, I find the mention of the # totally useless when I'm looking at git logs in a terminal or in an IDE. I can't click on it. It makes me read more words than I need to to understand what's going on. If it was at the end, I could at least just ignore it (like I do with the signature fluff -- which at least has a good reason to be there).

@donraab @nikhilnanivadekar @motlin (did I miss any committers? why can't see a list somewhere?) please comment.

P.S. We may have an issue with the wiki -- I found random SPAM in the footer which I fixed.

**Comment by nikhilnanivadekar** [comment]
@mohrezaei I agree with you about the difficulty of looking commit messages. I think the correct thing to do is to have “Closes #issueNumber “ at the end of the commit message. That way things are not obscured, at the same time, GitHub can automatically close the issue once the commit is merged.
Thoughts?

List of committers is found on the Project Details page of Eclipse Foundation, so, may be we should add a link to that page somewhere.

**Comment by mohrezaei** [comment]
@nikhilnanivadekar I found the committers [list](https://github.com/orgs/eclipse/teams/technology-collections-committers) in Eclipse organization on github (which is where the permissions are coming from). It's listed as a "secret", so only committers can see it.

Even though I'm a committer, I don't have admin permission on this repo (I don't see the "Settings" tab at the top). I wanted to change the wiki settings so we don't end up with SPAM by restricting writes to committers. I don't know who can do that... someone at Eclipse I guess.

**Comment by nikhilnanivadekar** [comment]
I don’t think any of us have the admin rights. May be we need to open a bugzilla for them to do it?

**Comment by motlin** [comment]
I gave the same feedback and a bunch of other feedback on the original
issue.

https://github.com/eclipse/eclipse-collections/issues/871

The whole page is so misleading and I haven't had time to fix it, so I
wouldn't mind temporarily taking it down. I don't know why, but I do have
permission to edit or delete the page.

**Comment by mohrezaei** [comment]
@motlin I'll quote your original recommendation here. It all makes sense, but looks like something got lost in translation:

> Any of Closes, Fixes, or Resolves is fine, depending on context. I would never add it at the beginning, always the end. I'd also add the context that you only need to close the issue if it already exists. You don't have to create a new issue just to close it with a PR. A PR is an issue (they share a namespace / sequence number)

Also, the problem with the wiki is that anyone with a Github account can do whatever they want (including add SPAM, which I cleaned up). We can't change those permissions without going through Eclipse foundation.

**Comment by mohrezaei** [comment]
Based on the consensus here, I edited the page. I don't think it's bad enough to take down, but I could be convinced otherwise :smile:

**Comment by nikhilnanivadekar** [comment]
@mohrezaei raised a bugzilla to restrict access rights: https://bugs.eclipse.org/bugs/show_bug.cgi?id=566112

In addition to that changed the content of Working with GitHub a little bit to be inline with Craig's review comments.
